### ISSUE TITLE
Categorical time sequence LSTM

### ISSUE THREAD

**Issue by max-white** [keras-team/keras#8315]
Hey there,

i converted my MLP Network to an LSTM Network to check if i can get a better accuracy taking
the time into consideration. I have sound signals that are split into multiple frames (87 Frames each sound signal) and 39 Features generated from those Frames (Mel Frequency Cepstrum Coefficients).
In my MLP Network i had a input dimension of 39 Feautres and in the LSTM Network i had to reshape
this vector to 3D Tensor and decided to have those 87 frames as timesteps. (It might make more sense to put this value higher).

So from `(215499, 39)` i reshaped the vector to `(2477, 87, 39)` .
my output vector `y` also had to be reshaped from `(215499,)` to `(2477, 87, 1)`
and i had to add the `TimeDistribution` Wrapper around my `Dense()` output layer because
of the many-to-many (categories) nature of my classification problem.

here is the split and reshape part:
```
num_classes = 3

timesteps = 87
n_features = 39

features = features.reshape((int(features.shape[0]/timesteps), timesteps, 39))
labels = labels.reshape((int(labels.shape[0]/timesteps), timesteps, 1))

train_test_split = np.random.rand(len(features)) < 0.50

x_train = features[train_test_split]
y_train = labels[train_test_split]
x_test = features[~train_test_split]
y_test = labels[~train_test_split]

y_train = keras.utils.to_categorical(y_train, num_classes)
y_test = keras.utils.to_categorical(y_test, num_classes)

y_train = y_train.reshape((int(y_train.shape[0]/timesteps), timesteps, num_classes))
y_test = y_test.reshape((int(y_test.shape[0]/timesteps), timesteps, num_classes))

```
code
the creation of the model:
```
model = Sequential()
model.add(LSTM(max_features, input_shape=(timesteps, n_features), return_sequences=True))
model.add(TimeDistributed(Dense(num_classes, activation='softmax')))
```

`features = features.reshape((int(features.shape[0]/timesteps), timesteps, 39))` depending
on the timesteps the dimensionalty of the 3D Tensor changes.

the keras function to categorical changes my 3D tensor back to 2D Tensor, so i have to reshape again.
Is there another way how tho accomplish this?

Greetings

**Comment by StanSilas** [comment]
Did you figure this out?

**Comment by max-white** [comment]
here is the working version:
```
num_classes = 3

#timesteps = time_range
timesteps = time_range # amount of timeperiods of the audiosignal defined by the short time fourier transformation

#features = features[timesteps]

features = features.reshape((int(features.shape[0]/timesteps), timesteps, 39)) # 2D -> 3D Vektor
labels = labels.reshape((int(labels.shape[0]/timesteps), timesteps, 1)) # 2D -> 3D Vektor

train_test_split = np.random.rand(len(features)) < 0.70

x_train = features[train_test_split]
y_train = labels[train_test_split]
x_test = features[~train_test_split]
y_test = labels[~train_test_split]

y_train = keras.utils.to_categorical(y_train, num_classes)
y_test = keras.utils.to_categorical(y_test, num_classes)

y_train = y_train.reshape((int(y_train.shape[0]/timesteps), timesteps, num_classes))
y_test = y_test.reshape((int(y_test.shape[0]/timesteps), timesteps, num_classes))
```
